pkg/services/alerting/engine_test.go

```diff
@@ -113,6 +200,29 @@ func TestEngineProcessJob(t *testing.T) {
 				engine.processJobWithRetry(context.TODO(), job)
 				So(evalHandler.CallNb, ShouldEqual, expectedAttempts)
 			})
+
+			Convey("pended alert for datasource -> result handler should be worked", func() {
+				// reduce alert timeout to test quickly
+				originAlertTimeout := alertTimeout
+				alertTimeout = 5 * time.Second
+				transportTimeoutInterval := 5 * time.Second
+				serverBusySleepDuration := 4 * time.Second
```

Comment:
I put a test for this case and these seconds are configurable.
Tell me if you want to make it shorter or have any suggestions.